x/programs/cmd/simulator/cmd/plan.go

```diff
@@ -328,13 +328,16 @@ func (c *runCmd) createCallParams(ctx context.Context, db state.Immutable, param
 			key := val
 			// get named public key from db
 			pk, ok, err := storage.GetPublicKey(ctx, db, val)
-			if ok {
-				// otherwise use the public key address
-				key = string(pk[:])
-			}
 			if err != nil {
 				return nil, err
 			}
+			if !ok && endpoint != EndpointKey {
+				// using not stored named public key in other context than key creation
+				return nil, fmt.Errorf("%s: %s", ErrNamedKeyNotFound, val)
+			}
+			if ok {
+				key = string(pk[:])
+			}
```

Comment:
It's interesting that this is silently failing in the case of the key not being found in the database, we should probably do something about this later

Reply:
of course 🤦 thanks